src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -325,10 +326,164 @@ module.exports = class LevelPlane extends Array {
       posToRefresh.push(this.redstoneListON[i]);
     }
 
+    // Once we're done updating redstoneWire states, check to see if doors and pistons should open/close.
+    for (let i = 0; i < this.length; ++i) {
+      this.getIronDoors(i);
+      this.getPistonState(i);
+    }
     return posToRefresh;
   }
 
   /**
+  * Evaluates what state Iron Doors on the map should be in.
+  */
+  getIronDoors(index) {
+    if (this[index].blockType === "doorIron") {
+      this[index].isPowered = this.powerCheck(this.indexToCoordinates(index));
+      if (this[index].isPowered && !this[index].isOpen) {
+        this.animateDoor(index, true);
+      } else if (!this[index].isPowered && this[index].isOpen) {
+        this.animateDoor(index, false);
+      }
+    }
+  }
+
+  /**
+  * Evaluates what state Pistons on the map should be in.
+  */
+  getPistonState(index) {
+    if (this[index].blockType.startsWith("piston") && this[index].blockType !== ("pistonArm")) {
+      this[index].isPowered = this.powerCheck(this.indexToCoordinates(index));
+      if (this[index].isPowered) {
+        this.activatePiston(this.indexToCoordinates(index));
+        if (this.levelModel) {
+          this.levelModel.controller.updateFowPlane();
+          this.levelModel.controller.updateShadingPlane();
+        }
+      } else if (!this[index].isPowered) {
+        this.deactivatePiston(this.indexToCoordinates(index));
+        if (this.levelModel) {
+          this.levelModel.controller.updateFowPlane();
+          this.levelModel.controller.updateShadingPlane();
+        }
+      }
```

Comment:
This can be simplified to:

```javascript
const coordinates = this.indexToCoordinates(index)

if (this[index].isPowered) {
  this.activatePiston(coordinates);
} else {
  this.deactivatePiston(coordinates);
}

if (this.levelModel) {
  this.levelModel.controller.updateFowPlane();
  this.levelModel.controller.updateShadingPlane();
}
```

Reply:
Although that does raise the question of whether or not we want to do all this work every time or only when `isPowered` changes